### PR TITLE
AddPowerSupport_tblib

### PR DESCRIPTION
**Pull request by santosh653**
Creating tblib.travis.yml with proposed changes.,

### CODE REVIEW

File: travis-ymls/tblib.travis.yml [hunk 1]
```diff
@@ -0,0 +1,98 @@
+# ----------------------------------------------------------------------------
+#
+# Package             : tblib
+# Source Repo         : https://github.com/ionelmc/python-tblib
+# Travis Job Link     : https://travis-ci.com/github/santosh653/pylibdmtx
```
Comment (gururajrkatti) [inline travis-ymls/tblib.travis.yml:5]: travis link pointing to https://travis-ci.com/github/santosh653/pylibdmtx where as project/repos is differnt

**Comment by santosh653** [comment]
@gururajrkatti I have corrected the error.,

**Comment by santosh653** [comment]
@gururajrkatti Can you please review & approve this?

**Comment by gururajrkatti** [comment]
I see two files here. Can you pl. clarify and correct

**Comment by santosh653** [comment]
@gururajrkatti I have deleted the old file which has errors, and retained the latest file,. Now only one file can be seen in Files Tab.
Kindly verify & approve.

**Comment by santosh653** [comment]
@gururajrkatti The hyperwiser link with name "Add Power Support" directs to latest File.,

**Comment by santosh653** [comment]
@gururajrkatti Can you please review?

**Comment by gururajrkatti** [comment]
Don't we have /lib/x86_64-linux-gnu/libSegFault.so for ppc64le ?

**Comment by santosh653** [comment]
@gururajrkatti Hi Guru, this variable(- LD_PRELOAD=/lib/x86_64-linux-gnu/libSegFault.so) is set upfront after which the code is running for the mentioned python versions, for power I have excluded "pypy" & "pypy3" as they are not supported.

**Comment by santosh653** [comment]
This variable seems not available on power & is ignored for power and the build went fine.

**Comment by gururajrkatti** [comment]
Ok. can you upload the new file

**Comment by santosh653** [comment]
@gururajrkatti .so file can't be uploaded here I guess, it's not supported. I'm not able to upload. Let me know if I am wrong.

**Comment by santosh653** [comment]
However I have uploaded the same at my fork of the package URL.,

File: travis-ymls/tblib.travis.yml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+#
+# Script License      : Apache License, Version 2 or later
+#
+# ----------------------------------------------------------------------------
+language: python
+dist: xenial
+cache: false
+env:
+  global:
+    - LD_PRELOAD=/lib/x86_64-linux-gnu/libSegFault.so
```
Comment (gururajrkatti) [inline travis-ymls/tblib.travis.yml:17]: I still see the file points to libSegFault

**Comment by santosh653** [comment]
@gururajrkatti Added travis file with required changes.,

File: travis-ymls/tblib.travis.yml [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+env:
+  global: SEGFAULT_SIGNALS=all
+    
+matrix:
+  include:
+    - python: '3.6'
+      env:
+        - TOXENV=check
+      arch: amd64
+      global: LD_PRELOAD=/lib/x86_64-linux-gnu/libSegFault.so
```
Comment (gururajrkatti) [inline travis-ymls/tblib.travis.yml:24]: global is not a valid keyword within matrix. Also the travis link is pointing to wrong path https://travis-ci.com/github/santosh653/pylibdmtx

**Comment by santosh653** [comment]
@gururajrkatti But the code is running fine, I have corrected the Travis link.,

**Comment by gururajrkatti** [comment]
You can see "unknown key global" in the travis link https://travis-ci.com/github/santosh653/python-tblib/jobs/491940985/config. Instead put the variable as part of env itself.

**Comment by santosh653** [comment]
@gururajrkatti Can you please review now with latest changes?

File: travis-ymls/tblib.travis.yml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+# ----------------------------------------------------------------------------
+language: python
+
+dist: xenial
+os: linux
+cache: false
+env:
+  global: 
+  - SEGFAULT_SIGNALS=all
+  - if [[ "$TRAVIS_CPU_ARCH" = "amd64" ]]; then
```
Comment (gururajrkatti) [inline travis-ymls/tblib.travis.yml:20]: if statement not valid in the env:

**Comment by santosh653** [comment]
@gururajrkatti required changes done, no more warnings or errors under Travis config.,

**Review by gururajrkatti — COMMENTED** [review]
I see error as below in the ppc build ```$ export LD_PRELOAD=/usr/lib64/ld64.so.2
ERROR: ld.so: object '/usr/lib64/ld64.so.2' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.``` Kindly check if the mentioned library exist in the corresponding dist

**Comment by santosh653** [comment]
@gururajrkatti, I see the "dist: Focal" on my local VM as below, & the Library(/usr/lib64/ld64.so.2) is present for this dist.
ubuntu@santoshkulkarni:~$ cat /etc/issue
Ubuntu 20.04.1 LTS \n \l
whereas the travis CI uses dist: Xenial & OS: linux by default 
That is why I am getting the error in Travis.

**Comment by gururajrkatti** [comment]
Pl. check if you find the library in xenial container and use the same in travis

**Comment by santosh653** [comment]
@gururajrkatti I am not sure about this How to change the distribution to Xenial from Focal on my VM as it is already showing dist as Focal.

**Comment by gururajrkatti** [comment]
@santosh653 as mentioned you can use xenial container.

**Comment by seth-priya** [comment]
@santosh653 any update on this?

**Comment by seth-priya** [comment]
@santosh653 any update on this?

**Comment by gururajrkatti** [comment]
@santosh653 any updates on this

**Comment by santosh653** [comment]
closing this PR

**Pull request closed without merging**